src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/sam/SamAuthProvider.scala

```diff
@@ -127,11 +129,15 @@ class SamAuthProvider(val config: Config, serviceAccountProvider: ServiceAccount
     */
   override def filterUserVisibleClusters(userInfo: UserInfo, clusters: List[(GoogleProject, ClusterName)])(implicit executionContext: ExecutionContext): Future[List[(GoogleProject, ClusterName)]] = {
     for {
-      owningProjects <- Future(blocking(samClient.listOwningProjects(userInfo)).toSet)
-      createdClusters <- Future(blocking(samClient.listCreatedClusters(userInfo)).toSet)
+      owningProjects <- retryUntilSuccessOrTimeout(shouldInvalidateSamCacheAndRetry, s"SamAuthProvider.owningProjects.filterUserVisibleClusters call failed")(samRetryInterval, samRetryTimeout) { () =>
+        Future(blocking(samClient.listOwningProjects(userInfo)).toSet)
+      }
+      createdClusters <- retryUntilSuccessOrTimeout(shouldInvalidateSamCacheAndRetry, s"SamAuthProvider.owningProjects.filterUserVisibleClusters call failed")(samRetryInterval, samRetryTimeout) { () =>
+        Future(blocking(samClient.listCreatedClusters(userInfo)).toSet)
+      }
     } yield {
       clusters.filter { case (project, name) =>
-        owningProjects.contains(project) || createdClusters.contains((project, name))
+        owningProjects.contains(project) || createdClusters.right.get._2.contains((project, name))
```

Comment:
I don't think `.right.get._2` is correct in this case. `retryUntilSuccessOrTimeout` returns a `RetryableFuture`. What you _want_ is this implicit conversion to kick in: https://github.com/broadinstitute/workbench-libs/blob/develop/util/src/main/scala/org/broadinstitute/dsde/workbench/util/Retry.scala#L98

However I don't think it is in this case. You may need to call it explicitly. This code might work:
```
  override def filterUserVisibleClusters(userInfo: UserInfo, clusters: List[(GoogleProject, ClusterName)])(implicit executionContext: ExecutionContext): Future[List[(GoogleProject, ClusterName)]] = {
    for {
      owningProjects <- retryableFutureToFuture(retryUntilSuccessOrTimeout(shouldInvalidateSamCacheAndRetry, s"SamAuthProvider.owningProjects.filterUserVisibleClusters call failed")(samRetryInterval, samRetryTimeout) { () =>
        Future(blocking(samClient.listOwningProjects(userInfo)).toSet)
      })
      createdClusters <- retryableFutureToFuture(retryUntilSuccessOrTimeout(shouldInvalidateSamCacheAndRetry, s"SamAuthProvider.owningProjects.filterUserVisibleClusters call failed")(samRetryInterval, samRetryTimeout) { () =>
        Future(blocking(samClient.listCreatedClusters(userInfo)).toSet)
      })
    } yield {
      clusters.filter { case (project, name) =>
        owningProjects.contains(project) || createdClusters.contains((project, name))
      }
    }
  }
```